src/app/cube/user-profile/user-profile.component.ts

```diff
@@ -35,6 +38,7 @@ export class UserProfileComponent implements OnInit {
     this.route.data.subscribe(val => {
```

Comment:
Be sure to unsubscribe from subscriptions when the component is destroyed.

Reply:
You can also avoid a subscription all together by grabbing the data from route snapshot if the data isn't expected to change during the lifetime of the component.